presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -748,4 +700,16 @@ static void addCompletionCallback(QueryExecution queryExecution, Runnable callba
             callback.run();
         }
     }
+
+    private QueryExecution requiredQuery(QueryId queryId)
```

Comment:
The naming is weird. What is a "required query"? How about naming these methods "getQuery" and "getQueryIfPresent". Or maybe just a getQuery that returns Optional<Query> and let the "get()" call handle the concept of "required".

Reply:
`getQuery` and `tryGetQuery`?

BTW we've used this naming convention somewhere else before (don't remember now).

Reply:
`MoreFutures` has `getFutureValue` and `tryGetFutureValue`.